packages/firestore/src/util/async_queue.ts

```diff
@@ -289,15 +289,6 @@ export class AsyncQueue {
           const message = error.stack || error.message || '';
           logError('INTERNAL UNHANDLED ERROR: ', message);
 
-          // Escape the promise chain and throw the error globally so that
-          // e.g. any global crash reporting library detects and reports it.
-          // (but not for simulated errors in our tests since this breaks mocha)
-          if (message.indexOf('Firestore Test Simulated Error') < 0) {
```

Comment:
This is still used, isn't it?

https://github.com/firebase/firebase-js-sdk/blob/bd85150b032ee360cb9e5e7c39b5a47de9b2bba6/packages/firestore/test/unit/util/async_queue.test.ts#L75

You mentioned in the PR description that you ran the integration tests... did you run the unit tests also?  Because it seems like the unit tests should fail with this line removed.

Reply:
This is special handling of test code that is no longer needed, since we are not using `mocha` anymore (`karma` runs our browser tests). So the same test passes now without this special handling.

As for running the unit tests - all tests run as part of CI, and I usually run them locally as well. There are some tests that are "harder" to run locally, such as the minified integration tests (see /integration/firestore), but even those are run in CI.